File: examples/python/streamable.py
import requests

API_BASE = 'https://cog.api.br/api/v1'

def download_streamable(url: str) -> dict:
    """
    Download video from Streamable
    
    Args:
        url: Streamable video URL
        
    Returns:
        dict: Download data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/streamable/download', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Streamable Video Downloaded Successfully!')
            print(f"Title: {data['data']['title']}")
            print(f"Quality: {data['data']['quality']}")
            print(f"Download URL: {data['data']['downloadUrl']}")
            print(f"Duration: {data['data']['duration']} seconds")
            print(f"Resolution: {data['data']['width']}x{data['data']['height']}")
            print(f"Filesize: {data['data']['filesize'] / 1024 / 1024:.2f} MB")
            
            return data['data']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_streamable_formats(url: str) -> list:
    """
    Get available formats for Streamable video
    
    Args:
        url: Streamable video URL
        
    Returns:
        list: Available formats or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/streamable/formats', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Available Formats:')
            for index, format_info in enumerate(data['formats'], 1):
                print(f'\nFormat {index}:')
                print(f"  ID: {format_info['formatId']}")
                print(f"  Quality: {format_info['quality']}")
                print(f"  Extension: {format_info['ext']}")
                print(f"  Size: {format_info['filesize'] / 1024 / 1024:.2f} MB")
            
            return data['formats']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_streamable_info(url: str) -> dict:
    """
    Get information about Streamable video
    
    Args:
        url: Streamable video URL
        
    Returns:
        dict: Video information or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/streamable/info', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Streamable Video Information:')
            print(f"Title: {data['info']['title']}")
            print(f"Duration: {data['info']['duration']} seconds")
            print(f"Description: {data['info']['description']}")
            print(f"Quality: {data['info']['quality']}")
            print(f"Filesize: {data['info']['filesize'] / 1024 / 1024:.2f} MB")
            
            return data['info']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

if __name__ == '__main__':
    print('=== Streamable Download Example ===\n')
    
    # Example 1: Download Streamable video
    streamable_url = 'https://streamable.com/example'
    print(f'Downloading from Streamable: {streamable_url}')
    download_streamable(streamable_url)
    
    print('\n=== Streamable Formats Example ===\n')
    
    # Example 2: Get available formats
    print(f'Getting formats for: {streamable_url}')
    get_streamable_formats(streamable_url)
    
    print('\n=== Streamable Info Example ===\n')
    
    # Example 3: Get video info
    print(f'Getting video info: {streamable_url}')
    get_streamable_info(streamable_url)

File: examples/nodejs/streamable.js
const axios = require('axios');

const API_BASE = 'https://cog.api.br/api/v1';

/**
 * Download video from Streamable
 * @param {string} url - Streamable video URL
 */
async function downloadStreamable(url) {
  try {
    const response = await axios.get(`${API_BASE}/streamable/download`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Streamable Video Downloaded Successfully!');
      console.log('Title:', response.data.data.title);
      console.log('Quality:', response.data.data.quality);
      console.log('Download URL:', response.data.data.downloadUrl);
      console.log('Duration:', response.data.data.duration, 'seconds');
      console.log('Resolution:', `${response.data.data.width}x${response.data.data.height}`);
      console.log('Filesize:', (response.data.data.filesize / 1024 / 1024).toFixed(2), 'MB');
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get available formats for Streamable video
 * @param {string} url - Streamable video URL
 */
async function getStreamableFormats(url) {
  try {
    const response = await axios.get(`${API_BASE}/streamable/formats`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Available Formats:');
      response.data.formats.forEach((format, index) => {
        console.log(`\nFormat ${index + 1}:`);
        console.log('  ID:', format.formatId);
        console.log('  Quality:', format.quality);
        console.log('  Extension:', format.ext);
        console.log('  Size:', (format.filesize / 1024 / 1024).toFixed(2), 'MB');
      });
      
      return response.data.formats;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get information about Streamable video
 * @param {string} url - Streamable video URL
 */
async function getStreamableInfo(url) {
  try {
    const response = await axios.get(`${API_BASE}/streamable/info`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Streamable Video Information:');
      console.log('Title:', response.data.info.title);
      console.log('Duration:', response.data.info.duration, 'seconds');
      console.log('Description:', response.data.info.description);
      console.log('Quality:', response.data.info.quality);
      console.log('Filesize:', (response.data.info.filesize / 1024 / 1024).toFixed(2), 'MB');
      
      return response.data.info;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

// Example usage
(async () => {
  console.log('=== Streamable Download Example ===\n');
  
  // Example 1: Download Streamable video
  const streamableUrl = 'https://streamable.com/example';
  console.log('Downloading from Streamable:', streamableUrl);
  await downloadStreamable(streamableUrl);
  
  console.log('\n=== Streamable Formats Example ===\n');
  
  // Example 2: Get available formats
  console.log('Getting formats for:', streamableUrl);
  await getStreamableFormats(streamableUrl);
  
  console.log('\n=== Streamable Info Example ===\n');
  
  // Example 3: Get video info
  console.log('Getting video info:', streamableUrl);
  await getStreamableInfo(streamableUrl);
})();

module.exports = {
  downloadStreamable,
  getStreamableFormats,
  getStreamableInfo
};
